### ISSUE TITLE
Authorization error response is missing State property value.

### ISSUE THREAD

**Issue by 12dinesh12** [IdentityServer/IdentityServer4#4719]
Hi, 

In case of error the authorisation error response is missing state parameter value.
OpenID documentation clearly explains that if state was present as part of request then it should be included in auth error response as well.

link:  https://openid.net/specs/openid-connect-core-1_0.html#AuthError

I think this needs to be fixed in https://github.com/IdentityServer/IdentityServer4/blob/c50458957cab9d39b2ad4938a77d38aa27f36dd7/src/IdentityServer4/src/Endpoints/AuthorizeEndpointBase.cs#L129

Please clarify.

Thanks
Dinesh

**Comment by brockallen** [comment]
It is: 

https://github.com/IdentityServer/IdentityServer4/blob/18897890ce2cb020a71b836db030f3ed1ae57882/src/IdentityServer4/src/Extensions/AuthorizeResponseExtensions.cs#L53

**Comment by 12dinesh12** [comment]
@brockallen : yes the condition is there but the value is always not passed. look below.
  
return new AuthorizeResult(new AuthorizeResponse
            {
                Request = request,
                Error = error,
                ErrorDescription = errorDescription,
                SessionState = request?.GenerateSessionStateValue()
            });

This is taken from the CreateErrorResultAsync method. my previous comment points to the exact line number. State is missing but SessionState is being added. It is definitely a bug. State should also be added while constructing the AuthResponse.

**Comment by 12dinesh12** [comment]
Also we are using version 2.3.2 of the library where request object validation happens much later in the flow. so if basic validations fails then it skips setting state value. Do you think the issue is only in that specific version?
